### ISSUE TITLE
Sort toots chronologically

### ISSUE THREAD

**Issue by woj-tek** [tootsuite/mastodon#7900]
It's kinda related/followup to #1539 but for someone who likes to catch-up with what happened in the order it happened it would nice to have chronological sort. But for it to work we would also require *mark as read* feature (so after opening we would only see new items at the top).

This is mostly a bit quirky, but not everyone spends all the time on mastodon and usually catches up every now and then so that would be great way to do it.

* * * *

- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.

**Comment by trwnh** [comment]
Mastodon is already sorted chronologically -- in fact, it's one of the defining features. What exactly do you mean by this?

**Comment by woj-tek** [comment]
OK, they are sorted chronologically but… I meant reversed, i.e. oldest on top. Currently we have the newest on top.

I know that this is quite unorthodox and every other "social" network does it like that but I usually operate on "get done" basis (open, read from oldest to newest, close): mail, rss and it would be great to be able to read toots like that as well.

**Comment by trwnh** [comment]
Ahhh, I see what you mean now. It sounds like you're asking for a more email-like client. I'm not sure that would work with the default Mastodon webapp's current model, but I'm sure there's demand for that. And it probably makes sense since ActivityPub is basically email-over-JSON.

**Comment by woj-tek** [comment]
Yes, something like that. I would think that quite a lot of people would like this *modus operandi* (with the exception of those using it as a chat app I guess).

**Comment by MaPePeR** [comment]
I feel like the biggest reason to have the _reversed_ chronological ordering is ease of implementation:
* Scrolling down/swiping up to get to new content is a lot more comfortable & common than scrolling up/swiping down.
  * Currently scrolling down/swiping up shows older content.
  * I'm not aware of any chat app that adds new posts at the top instead of the bottom
* It is a lot easier to have a website that is fixed at the top and can be scrolled down indefinitely to load more content than the other way around (Which is required for non-reversed chronological ordering)
* Chronological ordering requires storing where you left of (Compare #20956) (Ideally across clients)

To have good non-reversed chronological ordering mastodon would have to remember the last scroll position and then jump you to that position, so old posts can be reached by scrolling up and new posts by scrolling down and scrolling down stops when there is nothing new available.

Currently my _modus operandi_ is:
* Load Timeline
* Scroll down to read unread posts
  * Be annoyed that threads are reversed in order
* Continue until familiar posts come up
* Go back to top & Repeat

With a non-reversed chronological order that gets a lot better and reduces a lot of repeated content and the "have I seen this already"-guessing. (And it gets less slot-machiney, because you immediately know what is new and what is old content)
The "jump to the latest posts", even though you haven't seen the posts in-between should be a deliberate user-decision and not something that happens on accident, so the users are aware, that it's their own fault, when they read content in reverse order, because they _decided_ to jump down and are now are scrolling up.

Currently the "jump to the latest posts" is the default, which makes it annoying.

I think having a non-reversed chronological order would greatly alleviate the requirement for special handling for threads in the timeline (See #8615).

The only place where I see reverse-chronoligcal ordering making sense is a mail client, but that is no comparison to mastodon, because a mail client can show a lot more mails at once, filter by unread/read, sort mails into folders, and much more, which isn't possible in mastodon. (And most mail clients can detect threads of mails and then change the order of presenting the mails in a thread, so they are not reverse-chronological anymore)

**Comment by hakonalstadheim** [comment]
> To have good non-reversed chronological ordering mastodon would have to remember the last scroll position and then jump you to that position, so old posts can be reached by scrolling up and new posts by scrolling down and scrolling down stops when there is nothing new available.

Not necessarily. I'd be _very_ happy indeed just to have the newest toot in my timeline at the bottom of the screen, pushing older content out off the top of the window. That way at leas when there are several posts in a thread, you start reading at the top and down, without having to move your eyes to to find the next post, you just go on reading.

**Comment by MaPePeR** [comment]
I made an attempt at implementing a Web-Client that shows a chronological timeline.

Repository: https://github.com/MaPePeR/chrono-todon
Test-Page: https://mapeper.github.io/chrono-todon/

It's only barely a Proof of Concept. It does not support a lot of features, but it roughly does what I sketched above:

* Starts at the current [marker](https://docs.joinmastodon.org/methods/markers/) for your home-timeline
* Infinite-Scroll upwards to older content
* Loads new content when scrolling down
* Stops at newest content

So far I like it. But I'm not sure I will be able to continue working on it to make it actually usable.

**Comment by daemontus** [comment]
I would also like to support the time-based pagination idea. First of all, as others have suggested, I often want to read posts in the order in which they were posted, but I don't really open the app that often. So even being able to "scroll down to 12h ago" would be awesome. This just makes the most sense to me as a "catching up" mechanism unless I am just doom scrolling some random timeline. The other factor is that I often have a few days when I don't read any posts, so it would be amazing to be able to go back a few days, check what happened and then move on.

**Comment by VBoudry** [comment]
> I made an attempt at implementing a Web-Client that shows a chronological timeline.
> 
> Repository: [MaPePeR/chrono-todon](https://github.com/MaPePeR/chrono-todon) Test-Page: [mapeper.github.io/chrono-todon](https://mapeper.github.io/chrono-todon/)
> 
I just tried it.
As warned, it misses bells and whistles, but it behaves as I would expect for a chronological browsing.
Can this be implemented as an option ?

**Comment by Marzal** [comment]
Phanpy using the Cath-up option, is able to show your TL in chronological order.

**Comment by pedromj** [comment]
I am not sure what would be the best design, but having a non-reverse chronological order can help increasing the success of Mastodon.
